### ISSUE TITLE
Runway deployment broken?

### ISSUE THREAD

**Issue by jeffchuber** [calcom/synclinear.com#20]
<img width="851" alt="Screen Shot 2022-10-21 at 4 39 55 PM" src="https://user-images.githubusercontent.com/891664/197305448-890f7883-8552-498c-806b-5033095803cd.png">

I deployed the app to runway but Linear is telling me the URL for hooking it up is invalid. 

Here is an API key sanitized version

`https://linear.app/oauth/authorize?client_id=1234&redirect_uri=https://app-production-111.up.railway.app&scope=write&state=linear-asdf&response_type=code&prompt=consent`

I played around with the uri.. adding `/?` between the domain and the url params. As well as doing `encodeURIComponent` on the url, but I got the same thing from Linear. 

This could be a Linear bug! Unsure. 

Anyone else seeing this?

**Comment by tedspare** [comment]
Not your fault, @jeffchuber! This is a known issue that took a backseat. Thanks for bringing it up.

To self-host, you can either
1. manually generate API keys and populate the `LINEAR_API_KEY` and `GITHUB_API_KEY` env vars (solo use only) or
2. create your own OAuth apps under [Linear > Settings > API > Create new application](https://linear.app/neat-workspace/settings/api/applications/new) (with your Railway URL as the callback URL) and GitHub > your org > Settings > Developer > OAuth apps > New. Once created, you'll replace the `OAUTH_ID` fields in [constants](https://github.com/neat-run/linear-to-github/blob/main/utils/constants.ts).

Option 2 will allow you to add teammates later.

This limitation follows from Linear and GH OAuth disallowing wildcard callback URLs.

I'll correct this in the self-hosting docs!

**Comment by jeffchuber** [comment]
@tedspare thanks for the quick response :) 

Option 2 sounds great - your instructions are really good. One followup question - where can I find my "Railway URL" - is that just the base URL? eg `https://syncymcsyncface.up.railway.app/`? 

Is that URL also inputted into Github's `Authorization callback URL`? 

I am sure this is obvious to some. Appreciate the help!

**Comment by tedspare** [comment]
@jeffchuber great to hear! Thanks for catching a few unclear points (not obvious!).

The callback URL should actually be `https://projectname.up.railway.app/api` (key addition: `/api`). I'll clarify this in the docs.

And yep, GitHub OAuth takes that same callback URL.

<sub>:joy: `syncyMcsyncface`</sub>

**Comment by jeffchuber** [comment]
Perfect, that fixed things up perfectly. 

Thanks for the help!

**Comment by jeffchuber** [comment]
In the end I was unable to get this to work sadly. Whenever I connected with Github it would reroute me back to synclinear.com/***** instead of my app, I don't know why. Whenever I connected with Linear, it would say that my uri was invalid. Very strange. It's almost like it wasn't even deploying my code... but I did do a smoke test and change some copy on the home page to verify and it did show my updated copy. The project had no string "synclinear" that wasn't in docs or marketing copy. Very very odd - but ran out of time this time to make it work.
